lib/register_client.rb

```diff
@@ -96,9 +96,10 @@ def download_rsf(register, phase)
     end
 
     def parse_rsf(rsf)
-      items = []
+      items = {}
       entries = { user: [], system: [] }
       records = { user: {}, system: {} }
+      entry_number = 1
 
       rsf.each_line do |line|
```

Comment:
An alternative to this combined with `entry_number` is `rsf.each_line.with_index do |line, index| ...` and use `index + 1`.